patches/check_puya_patch

```diff
@@ -2,7 +2,7 @@
 
 PATCH=puya_core2_4_x.patch
 
-for DIR in ~/.platformio/packages/framework-arduinoespressif8266*/; do
+for DIR in ~/.platformio/packages/framework-arduinoespressif8266/ ~/.platformio/packages/framework-arduinoespressif8266*/; do
```

Comment:
Why this change?
Shouldn't `*` also match the empty chars?

Reply:
I didn't touch that file or line... at least I can't find it in the Changed files listing, and I have no intention of changing it 🤔

Reply:
Maybe it was already part of the just merged Actions build.
I will check to see if it really makes a difference.

Reply:
It is part of #3525